master/buildbot/worker/docker.py

```diff
@@ -157,9 +165,17 @@ def checkConfig(self, name, password, docker_host, image=None, command=None,
                     continue
 
     @defer.inlineCallbacks
-    def reconfigService(self, name, password, docker_host, image=None, command=None,
-                        volumes=None, dockerfile=None, version=None, tls=None, followStartupLogs=False,
-                        masterFQDN=None, hostconfig=None, autopull=False, alwaysPull=False, **kwargs):
+    def reconfigService(self, name, password, docker_host, image=None,
+                        command=None, volumes=None, dockerfile=None,
+                        version=None, tls=None, followStartupLogs=False,
+                        masterFQDN=None, hostconfig=None, autopull=False,
+                        alwaysPull=False, custom_context=False,
+                        encoding='gzip', buildargs=None, **kwargs):
+        """
+        :param custom_context: Bool, indicates the docker context is custom
```

Comment:
I had master/docs/manual/configuration/workers-docker.rst in mind when talking about docs. That page needs rewriting anyway, so I could move the docs of new parameters there when rewriting.